server/server.go

```diff
@@ -69,6 +78,115 @@ type Server struct {
 	handler           http.Handler
 }
 
+// LayoutBuilder is responsible for building Layouts
```

Comment:
I'd like to see all the builder interfaces and structs in a new file called builders.go